appcelerator-archive/amp
# Auto-login for signup when cluster has `--registration=none`

resolves: #1261 

adds the flag `--autologin` to automatically login user. Default value of the flag is true.

to test: 

Cluster with no registration 
```
$ make build
$ amp -s localhost cluster create --tag=local
$ amp -s localhost user signup --name=jgrahamjones --email=jgrahamjones@axway.com --password=Astrongpassword1
Verification is not necessary for this cluster.
Hi jgrahamjones! You have been automatically logged in.
$ amp -s localhost user ls
[localhost:50101]
USERNAME       EMAIL                    CREATED ON
jgrahamjones   jgrahamjones@axway.com   09 Jun 17 16:52
su             super@user.amp           09 Jun 17 16:51
$ amp -s localhost whoami
[localhost:50101]
Logged in as user: jgrahamjones
$ amp -s localhost user rm jgrahamjones
$ amp -s localhost user signup --name=jgrahamjones --email=jgrahamjones@axway.com --password=Astrongpassword1 --autologin=false
Verification is not necessary for this cluster.
Hi jgrahamjones! Please log in with your details using `amp login`.
$
```

Cluster with registration
```
$ make build
$ amp -s localhost cluster create --tag=local
$ amp -s localhost user signup --name=jgrahamjones --email=jgrahamjones@axway.com --password=Astrongpassword1
Hi jgrahamjones! Please check your email to complete the signup process.
$
```

`platform/tests` have been updated to remove any `amp login` commands as the default value for the `amp signup --autologin` is true.
